cirq-core/cirq/circuits/circuit.py

```diff
@@ -2306,7 +2309,55 @@ def _resolve_operations(
     return resolved_operations
 
 
+def _get_moment_annotations(
+    moment: 'cirq.Moment',
+) -> Iterator['cirq.Operation']:
+    for op in moment.operations:
+        if op.qubits:
+            continue
+        op = op.untagged
+        if isinstance(op, ops.GlobalPhaseOperation):
+            continue
+        if isinstance(op, CircuitOperation):
+            for m in op.circuit:
+                yield from _get_moment_annotations(m)
```

Comment:
IIRC, CircuitOperations already generate their own subdiagram when displayed as part of a larger circuit. Will this result in the qubit-less ops of the CircuitOperation being displayed multiple times? (Once for each nested CircuitOperation containing the qubit-less op)

Reply:
I considered this and tested that it works correctly. Unfortunately, I can't include the "circuit operation has more than just annotations" test because of https://github.com/quantumlib/Cirq/issues/4240 .

Reply:
In the CircuitOperation tests, I just included `hash(circuitop)` in the expected diagram string. If this is working as intended, I won't press for it here.